vector.go

```diff
@@ -23,6 +23,8 @@ type VectorField interface {
 	Dims() int
 	// Similarity metric to be used for scoring the vectors
 	Similarity() string
+	// index type - tuned for either latency/recall
+	IndexType() string
```

Comment:
Very poor choice of naming - please change to `IndexOptimization` or some other.